Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,31 @@
+# create a lock using `flock (2)`. A name is required as first argument.
+# the lock will be automatically unlocked when the shell process quits.
+lock() {
+  local name="$1"
+  local lock_file="/tmp/homebrew-${name}.lock"
```

Comment:
I think both these options are not optimal. Using `HOMEBREW_LIBRARY` or something similar is more likely to break due to permission issues. Including `HOMEBREW_REPOSITORY` in `lock_file` can lead to pretty long and ugly lock file names. I think the latter comes with fewer problems, so I'd prefer that.